### PR TITLE
Fix fuzz_pubkey on valid RSA keys

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
`fuzz_pubkey` was systematically failing on valid RSA keys, due to a bug in the sanity checks performed as part of the fuzzed code. Fix this.

Also improve the sanity checks on non-RSA keys.

### CODE REVIEW

File: programs/fuzz/fuzz_pubkey.c [hunk 1]
```diff
@@ -21,7 +21,10 @@ int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
             mbedtls_mpi_init( &DQ ); mbedtls_mpi_init( &QP );
 
             rsa = mbedtls_pk_rsa( pk );
-            if ( mbedtls_rsa_export( rsa, &N, &P, &Q, &D, &E ) != 0 ) {
+            if ( mbedtls_rsa_export( rsa, &N, NULL, NULL, NULL, &E ) != 0 ) {
```
Comment (sbutcher-arm) [inline programs/fuzz/fuzz_pubkey.c]: Minor - Pre-existing, but there's a mix of brace styles, with line 15.

We really need to check style in the CI.

File: programs/fuzz/fuzz_pubkey.c [hunk 2]
```diff
@@ -41,13 +41,20 @@ int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
 #if defined(MBEDTLS_ECP_C)
         if( mbedtls_pk_get_type( &pk ) == MBEDTLS_PK_ECKEY )
         {
-            mbedtls_ecp_keypair *ecp;
+            mbedtls_ecp_keypair *ecp = mbedtls_pk_ec( pk );
```
Comment (sbutcher-arm) [inline programs/fuzz/fuzz_pubkey.c]: Minor - Pre-existing, but mixing brace style.

File: programs/fuzz/fuzz_privkey.c [hunk 3]
```diff
@@ -44,7 +44,8 @@ int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
         else
 #endif
 #if defined(MBEDTLS_ECP_C)
-        if( mbedtls_pk_get_type( &pk ) == MBEDTLS_PK_ECKEY )
+        if( mbedtls_pk_get_type( &pk ) == MBEDTLS_PK_ECKEY ||
```
Comment (sbutcher-arm) [inline programs/fuzz/fuzz_privkey.c]: Minor - Pre-existing but mixing brace styles.

**Comment by sbutcher-arm** [comment]
@gilles-peskine-arm - This PR is approved and ready to go, except failing CI because you forgot to sign-off the commits.

**Comment by gilles-peskine-arm** [comment]
@sbutcher-arm There's no legal requirement for a signed-off-by line from Arm employees (or from people who've signed a CLA). We'll enforce it from everyone in the future, but we aren't enforcing it when not legally required yet.

**Comment by gilles-peskine-arm** [comment]
CI only fails on Mbed OS for an unrelated issue (importer).

**Pull request merged**